File: lib/schwarznasenschaf/admin/commands/clear.rb
# frozen_string_literal: true

module Schwarznasenschaf::Admin::Commands::Clear
  extend Discordrb::Commands::CommandContainer

  command_attributes = {
    description: 'Clears the last [x] messages from the channel',
    help_available: true,
    max_args: 1,
    min_args: 1,
    rescue: 'An  error occured while trying to execute this command.',
    # required_roles: [Schwarznasenschaf::Config::ROLES[:mod_team]],
    usage: 'clear [number]'
  }

  command :clear, command_attributes do |event, num_to_delete|
    result = Schwarznasenschaf::Admin::Commands.bulk_delete event.channel, num_to_delete.to_i
    Schwarznasenschaf::Admin::Commands.send_and_delete event.channel, result
  end
end

File: packages/bot/src/commands/admin/clear.js
import Config from 'config'
import { bulkDeleteMessages, sendAndDeleteMessage } from 'utils/commands'

export default class ClearCommand {
  constructor() {
    this.name = 'clear'
    this.description = 'Clears the last [x] messages from the channel'
    this.maxArgs = 1
    this.minArgs = 1
    this.requiredRoles = [Config.roles.modTeam]
    this.usage = `${Config.prefix}${this.name} [number]`
  }

  execute(message, args) {
    if (args[0]) {
      const numberMessagesToDelete = parseInt(args[0])

      if (Number.isNaN(numberMessagesToDelete)) {
        message.channel.send(`${args[0]} is not a number`)
        return
      }

      if (bulkDeleteMessages(message.channel, numberMessagesToDelete)) {
        sendAndDeleteMessage(`Successfully deleted ${numberMessagesToDelete} messages`, message.channel, 5)
      }
    }
  }
}
